OutOfSchool/OutOfSchool.DataAccess/Repository/ExternalImageStorage.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using MongoDB.Bson;
+using MongoDB.Driver.GridFS;
+using OutOfSchool.Services.Common.Exceptions;
+using OutOfSchool.Services.Contexts;
+
+namespace OutOfSchool.Services.Repository
+{
+    public class ExternalImageStorage : IExternalImageStorage
+    {
+        private readonly IGridFSBucket gridFsBucket;
+
+        public ExternalImageStorage(MongoDb db)
+        {
+            gridFsBucket = db.GetContext();
+        }
+
+        public async Task<Stream> GetByIdAsync(string pictureId)
+        {
+            try
+            {
+                return await gridFsBucket.OpenDownloadStreamAsync(new ObjectId(pictureId));
+            }
+            catch (Exception ex)
+            {
+                throw new ImageStorageException(ex);
+            }
+        }
+
+        public async Task<string> UploadImageAsync(Stream contentStream, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                contentStream.Position = uint.MinValue;
+                var objectId = await gridFsBucket.UploadFromStreamAsync(Guid.NewGuid().ToString(), contentStream, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
not relevant if we move to MySQL storage, but just a note (maybe we return to Mongo). Maybe leave a comment here.
Mongo default object ids are not unique and can be predicted (not as easy as integer but still). We can search by file name and use it as an identifier as uuid is pretty unique :)